node/core/pvf/src/pvf.rs

```diff
@@ -50,7 +51,7 @@ impl Pvf {
 	}
 
 	/// Returns the artifact ID that corresponds to this PVF.
-	pub(crate) fn as_artifact_id(&self) -> ArtifactId {
-		ArtifactId::new(self.code_hash)
+	pub(crate) fn as_artifact_id(&self, executor_params_hash: ExecutorParamsHash) -> ArtifactId {
+		ArtifactId::new(self.code_hash, executor_params_hash)
```

Comment:
nit: Do we still need this fn? If we have the `ExecutorParamsHash` then we can construct the new `PvfWithExecutorParams` struct. And call `as_artifact_id` on that.

Not sure if that makes sense in every situation, but I noticed a few places where we pass the PVF and params around separately where they could be consolidated, e.g. in `ToQueue::Enqueue` etc. (I could be missing something, haven't finished looking at the changes in the PVF module yet.)

Reply:
It seemed legit, and I've implemented it; it led to a somewhat more extensive code rework than I expected, but I feel it's much more consistent now.
However, I didn't push it to the end. It seems like after all the changes made, we're ready to get rid of the `Pvf` structure altogether in favor of `PvfWithExecutorParams` (or, the other way round, include `executor_params` info `Pvf` structure). To achieve that, additional changes to PVF host subsys and candidate validation subsys should be made. But I want to refrain from undertaking some global refactoring in the scope of this PR and this task. If the current solution is accepted, I'll file a separate issue.